src/main/java/org/roaringbitmap/RoaringBitmap.java

```diff
@@ -1265,6 +1265,51 @@ public boolean equals(Object o) {
   }
 
   /**
+   * Returns true if the other bitmap has no more than tolerance bits
+   * differing from this bitmap. The other may be transformed into a bitmap equal
+   * to this bitmap in no more than tolerance bit flips if this method returns true.
+   *
+   * @param other the bitmap to compare to
+   * @param tolerance the maximum number of bits that may differ
+   * @return true if the number of differing bits is smaller than tolerance
+   */
+  public boolean isHammingSimilar(RoaringBitmap other, int tolerance) {
+    final int size1 = highLowContainer.size();
+    final int size2 = other.highLowContainer.size();
+    int pos1 = 0;
+    int pos2 = 0;
+    int budget = tolerance;
+    while(budget >= 0 && pos1 < size1 && pos2 < size2) {
+      final short key1 = this.highLowContainer.getKeyAtIndex(pos1);
+      final short key2 = other.highLowContainer.getKeyAtIndex(pos2);
+      Container left = highLowContainer.getContainerAtIndex(pos1);
+      Container right = other.highLowContainer.getContainerAtIndex(pos2);
+      if(key1 == key2) {

```

Comment:
I don't know if I fully understand. Are you suggesting that if the keys aren't equal, for the sake of argument, key1 < key2 we should advance pos1 until key1 == key2, and decrement budget by left.cardinality for each container? 

I guess the idea behind was that most bitmaps aren't hamming similar, and it's better to get out as soon as the budget for differences is exhausted, hence a position by position by iteration. getCardinality is very cheap though so it might be a bit faster, if I have interpreted the term "gallop" correctly.